lib/cli/options.js

```diff
@@ -321,14 +322,38 @@ const loadOptions = (argv = []) => {
     args.opts = false;
     args._ = args._.concat(optsConfig._ || []);
   }
+  // special case: "extension" option should not combine with default value.
+  // normally we want to combine "array"-type options, and we _do_ with "extension", but only
+  // within user-defined configuration (args or anything else).
+  // we must also search through any aliases of "extension" because while the arguments are ///
+  // normalized by this point, the config file values are not.
+  // only the "canonical" option name is used in `mocharc`, so we needn't worry about clearing
+  // multiple options.
+  // NOTE: as of this writing, "extension" is the only default value which is of an "array" type;
+  // it's unknown whether the the below strategy should be generalized to any other future
+  // "array"-type default option.
+  const processedMocharc = Object.assign({}, mocharc);
```

Comment:
Deep clone, or see if this (e.g., ES7 `_.omit`) is supported...
```js
  let key = 'extension';
  const { [key]: _ignore, ...processedMocharc } = mocharc;
```

`Object.assign` here does shallow clone (after shallow freeze).
Surprised the `delete` didn't throw `TypeError`.